app/parse.py

```
@@ -15,8 +29,106 @@ class Product:
+    try:
+        wait = WebDriverWait(driver, 15)
+        accept_cookie = wait.until(
+            EC.element_to_be_clickable((
+                By.CLASS_NAME, "acceptCookies"
+            ))
+        )
+
+        accept_cookie.click()
+        time.sleep(5)
+    except TimeoutException:
+        print("cookie doesn't found")
+
+
+def scroll_more(driver: webdriver) -> None:
+    while True:
+        try:
+            wait = WebDriverWait(driver, timeout=15)
+            more = wait.until(
+                EC.element_to_be_clickable(
+                    (
+                        By.CLASS_NAME, "ecomerce-items-scroll-more")
+                )
+            )
+            more.click()
+
+        except Exception:
+            break
+
+
+def get_one_product(driver: webdriver) -> List[Product]:
+    elements = driver.find_elements(By.CLASS_NAME, "thumbnail")
+    arr = []
+    for element in elements:
+        title = element.find_element(
+            By.CLASS_NAME, "title"
+        ).get_attribute("title")
+        description = element.find_element(By.CLASS_NAME, "description").text
+        price = float(
+            element.find_element(By.CLASS_NAME, "price")
+            .text.replace("$", "")
+            .strip()
+        )
+        rating = len(element.find_elements(By.CLASS_NAME, "ws-icon-star"))
+        num_of_reviews = int(
+            element.find_element(By.CLASS_NAME, "review-count")
+            .text.split(" ")[0]
+        )
+        arr.append(
+            Product(
+                title=title,
+                description=description,
+                price=price,
+                rating=rating,
+                num_of_reviews=num_of_reviews
+            )
+        )
+    return arr
+
+
+def handle_page(filename: str, url: str) -> None:
+    driver = get_driver()
+    driver.get(url)
+    cookies(driver)
+    scroll_more(driver)
+    output_info = get_one_product(driver)
+    write_products_to_csv(filename, output_info)
```

Critical: `handle_page` creates a WebDriver but never quits it. This will leak browser processes across runs and cause CI/timeouts. Wrap the driver usage in try/finally (or a context manager) and call `driver.quit()` in the finally block so the driver is always terminated even on errors. Also ensure any exceptions are propagated or handled appropriately so failures are visible (don't silently swallow them).